src/Data/ExistsRE.purs

```diff
@@ -0,0 +1,20 @@
+module Data.ExistsRE where
+
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A variant of `Exists` that works for type constructors that accept a _row_
+-- | of effects.
+newtype ExistsRE (f :: # ! -> *) = ExistsRE (forall eff. f eff)
```

Comment:
Is this actually correct? I believe the quantification is now in the wrong place. This is saying that the content of the existential is a global section / limit of `f`, but what you really want is a colimit. I'm not sure if the following is possible in PureScript, but what you need is something like:

```
data ExistsRE (f :: # ! -> *) = forall eff. ExistsRE (f eff)
```

Alternatively, if this is the only way to implement the existential in PureScript, you'll need to hide the constructor of `ExistsRE`.